sql: clean up comments when swapping primary keys

fqazi opened the pull request:
Fixes: #71984

Previously, when swapping out primary keys comments
on old indexes would be leftover. This was inadequate because
entries would be leftover in the system.comments table
that could prevent SHOW CREATE  to fail when it attempts
to map comments for dropped indexes. To address this, this patch
will clean up comments on primary key swaps.

Release note (bug fix): Comments are not cleaned up when the table
primary keys are swapped, which can cause SHOW TABLE to fail.

fqazi committed:
sql: clean up comments when swapping primary keys

Fixes: #71984

Previously, when swapping out primary keys comments
on indexes could be leftover. This was inadequate because
entries would be leftover in the system.comments table
that could prevent SHOW CREATE  to fail when it attempts
to map comments for dropped indexes. To address this, this patch
will clean up comments on primary key swaps.

Release note (bug fix): Comments are not cleaned up when the table
primary keys are swapped, which can cause SHOW TABLE to fail.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/76277)
<!-- Reviewable:end -->

fqazi committed:
sql: SHOW CREATE did not show constraint comments

Previously, when using the SHOW CREATE command we
did not show comments create on constraints. To address
this, this patch will extend SHOW CREATE to show constraint
comments. Additionally, this patch addresses an issue where
on primary key swaps constraint IDs were unintentionally
reassigned.

Release note: None

ajwerner reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 3 of 39 files at r2, 4 of 6 files at r3.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/76277#-:-MvZ7JDs-TZ5Au87Tz-H:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @postamar)



<!-- Sent from Reviewable.io -->

fqazi commented:
@ajwerner TFTR!

fqazi commented:
bors r+
